### PR TITLE
CCO-105: Support gcp workload identity federation

### PR DESCRIPTION
**Pull request by akhil-rane**
To enable short-lived credentials in gcp cluster using workload identity
federation (similar to AWS STS), we need to make following set of changes

- update golang.org/x/oauth2 to a version that supports workload identity
- add kubernetes service account names to gcp credentials request manifest
  which can be used by ccoctl tool to create gcp service accounts with
  limited access

ref: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
x-ref: https://issues.redhat.com/browse/CCO-105

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -43,12 +43,14 @@ require (
...
+	google.golang.org/api v0.30.0
 	gopkg.in/yaml.v2 v2.4.0
 	k8s.io/api v0.21.0
 	k8s.io/apimachinery v0.21.0
 	k8s.io/client-go v0.21.0
 	k8s.io/klog/v2 v2.8.0
 	k8s.io/utils v0.0.0-20210305010621-2afb4311ab10
 )
+
+replace google.golang.org/grpc => google.golang.org/grpc v1.29.1
```
Comment (dmage) [inline go.mod:56]: Can't you use the latest version?

Comment (akhil-rane) [reply go.mod:58]: `github.com/openshift/library-go` has hard dependency on this version

**Comment by akhil-rane** [comment]
@dmage this looks good for a review. PTAL

**Comment by dmage** [comment]
@akhil-rane do you have a Jira issue for this?

**Comment by akhil-rane** [comment]
> @akhil-rane do you have a Jira issue for this?

yep https://issues.redhat.com/browse/CCO-105

**Comment by dmage** [comment]
/retitle CCO-105: Support gcp workload identity federation

**Pull request merged**